shared/triage_metadata_test.go

```diff
@@ -0,0 +1,227 @@
+}
+
+func TestAddToFiles_AddNewMetadataResult(t *testing.T) {
+	var amendment MetadataResults
+	json.Unmarshal([]byte(`{
+		"/foo/foo1/a.html": [
+			{
+				"url": "foo",
+				"product": "chrome",
+				"results": [
+					{"status": 6 }
+				]
+			}
+		]
+	}`), &amendment)
+
+	var path = "foo/foo1"
+	var fileMap = make(map[string]Metadata)
+	fileInBytes := []byte(`
+links:
+  - product: chrome
+    url: foo
+    results:
+    - test: b.html
+  - product: firefox-2
+    url: https://bug.com/item
+    results:
+    - test: b.html
+      subtest: Something should happen
+      status: FAIL
+    - test: c.html
+`)
+	var file Metadata
+	yaml.Unmarshal(fileInBytes, &file)
+	fileMap[path] = file
+
+	actualMap := addToFiles(amendment, fileMap, NewNilLogger())
+
+	assert.Equal(t, 1, len(actualMap))
+	actualInBytes, ok := actualMap["foo/foo1"]
+	assert.True(t, ok)
+
+	var actual Metadata
+	yaml.Unmarshal(actualInBytes, &actual)
+	assert.Equal(t, 2, len(actual.Links))
+	assert.Equal(t, "chrome", actual.Links[0].Product.BrowserName)
+	assert.Equal(t, "foo", actual.Links[0].URL)
+	assert.Equal(t, 2, len(actual.Links[0].Results))
+	assert.Equal(t, "b.html", actual.Links[0].Results[0].TestPath)
+	assert.Equal(t, "a.html", actual.Links[0].Results[1].TestPath)
+	assert.Equal(t, TestStatusFail, *actual.Links[0].Results[1].Status)
+	assert.Equal(t, "firefox", actual.Links[1].Product.BrowserName)
+	assert.Equal(t, "https://bug.com/item", actual.Links[1].URL)
+}
+
+func TestAddToFiles_AddNewMetadataLink(t *testing.T) {
+	var amendment MetadataResults
+	json.Unmarshal([]byte(`{
+		"/foo/foo1/a.html": [
+			{
+				"url": "foo1",
+				"product": "chrome",
+				"results": [
+					{"status": 6 }
+				]
+			}
+		]
+	}`), &amendment)
+
+	var path = "foo/foo1"
+	var fileMap = make(map[string]Metadata)
+	fileInBytes := []byte(`
+links:
+  - product: chrome
+    url: foo
+    results:
+    - test: b.html
+  - product: firefox-2
+    url: https://bug.com/item
+    results:
+    - test: b.html
+      subtest: Something should happen
+      status: FAIL
+    - test: c.html
+`)
+	var file Metadata
+	yaml.Unmarshal(fileInBytes, &file)
+	fileMap[path] = file
+
+	actualMap := addToFiles(amendment, fileMap, NewNilLogger())
+
+	assert.Equal(t, 1, len(actualMap))
+	actualInBytes, ok := actualMap["foo/foo1"]
+	assert.True(t, ok)
+
+	var actual Metadata
+	yaml.Unmarshal(actualInBytes, &actual)
+	assert.Equal(t, 3, len(actual.Links))
+	assert.Equal(t, "chrome", actual.Links[0].Product.BrowserName)
+	assert.Equal(t, "foo", actual.Links[0].URL)
+	assert.Equal(t, 1, len(actual.Links[0].Results))
+	assert.Equal(t, "b.html", actual.Links[0].Results[0].TestPath)
+	assert.Equal(t, "firefox", actual.Links[1].Product.BrowserName)
+	assert.Equal(t, "https://bug.com/item", actual.Links[1].URL)
+	assert.Equal(t, "chrome", actual.Links[2].Product.BrowserName)
+	assert.Equal(t, "foo1", actual.Links[2].URL)
+}
+
+func TestGenerateRandomInt(t *testing.T) {
```

Comment:
Uh.... this will fail, occasionally. Actually, it's very unlikely to fail, but it might.

It's not worth testing random number generation. What you could test is `getNewCommitBranchName`, with appropriate mocking of github.